packages/core/src/browser/shell/tab-bar-toolbar.tsx

```diff
@@ -187,6 +187,24 @@ export interface TabBarToolbarItem {
 
 }
 
+/**
+ * Tab-bar toolbar item backed by a `React.ReactNode`.
+ * Unlike the `TabBarToolbarItem`, this item is not connected to the command service.
+ */
+export interface ReactTabBarToolbarItem {
+    readonly id: string;
+    readonly node: React.ReactNode;
```

Comment:
How can one trigger rerendering?

Reply:
I do not think I'd need it, but I can add it to the API.

Any suggestions, by the way?

Reply:
If you would like to render it differently depending on user input. As we do for usual items in the search widget.

>  Any suggestions, by the way?

There has to be an event like `onDidChange` and then it should not be `node` but a function which produces a new node, i.e. `render(): React.ReactNode`.